base/set.jl

```diff
@@ -427,6 +427,37 @@ allunique(::Union{AbstractSet,AbstractDict}) = true
 
 allunique(r::AbstractRange) = !iszero(step(r)) || length(r) <= 1
 
+"""
+    allequal(itr) -> Bool
+
+Return `true` if all values from `itr` are equal when compared with [`isequal`](@ref).
+
+See also: [`unique`](@ref), [`allunique`](@ref), [`issorted`](@ref).
+
+# Examples
+```jldoctest
+julia> allequal([])
+true
+
+julia> allequal([1])
+true
+
+julia> allequal([1, 1])
+true
+
+julia> allequal([1, 2])
+false
+
+julia> allequal(Dict(:a => 1, :b => 1))
+false
+```
+"""
+allequal(itr) = isempty(itr) ? true : all(isequal(first(itr)), itr)
+
+allequal(c::Union{AbstractSet,AbstractDict}) = length(c) <= 1
```

Comment:
I'm not sure whether it is right to add this specialization for `AbstractSet` and `AbstractDict`, since it could be wrong for multisets and multidicts (it is currently wrong with respect to the `Multiset` object of package `Multisets.jl`). All the more so because I can't think of any interest of the `allequal` functions on dicts and sets except for the multiset and multidict kinds! Also it can error if `length` is not defined, which is not a requirement for `AbstractSet` and `AbstractDict` AFAIK.
If you agree, it can still be kept for the relevant Base cases of course (`Set`, `IdSet`, `KeySet`, `BitSet`; `Dict`, `IdDict`, `Pairs`, `WeakKeyDict`, `ImmutableDict`, `EnvDict`).

Reply:
I understand your point, but I'm not sure whether it makes sense for a `Multiset` to be a subtype of `AbstractSet`. In fact, `Multiset` in Multisets.jl was not a subtype of `AbstractSet` until the recent 0.4.0 version (the current version is 0.4.2).

I just modeled the `allequal` method for `AbstractSet` and `AbstractDict` after the `allunique` method that already exists:

```julia
allunique(::Union{AbstractSet,AbstractDict}) = true
```

As you can see, the existing `allunique` method is incorrect for multisets. So I'm not sure what the best course of action is.

Reply:
Oh that's a very good point. Being consistent is probably more important than just one edge-case, so I will try to remember submitting a PR to `Multisets` once this PR is merged then, that should take care of the issue. Thanks!

Reply:
I wish we had in `Base` a supertype of `AbstractDict` to include multisets, say `Associative`, and similarly for sets. But I otherwise agree that `AbstractSet` generally assumes uniqueness of keys.